src/util.test.js

```diff
@@ -0,0 +1,845 @@
+        expect.assertions(1);
+        expect(() => OskariMock.util.alterBrightness()).toThrowError(TypeError);
+    });
+});
+
+describe('keyExists function', () => {
+
+    const object = {
+        "level1": {
+            "level2": "value"
+        }
+    }
+
+    test('returns true when key exists', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.keyExists(object, "level1.level2")).toEqual(true);
+    });
+
+    test('returns false when key not exists', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.keyExists(object, "level1.dummy")).toEqual(false);
+    });
+
+    test('throws TypeError when parameters are not provided', () => {
+        expect.assertions(1);
+        expect(() => OskariMock.util.keyExists()).toThrowError(TypeError);
+    });
+});
+
+describe('naturalSort function', () => {
+
+    /*
+    * Natural sort returns 1 if parameters are in correct order with used ascending / descending and
+    * -1 if parameters are in incorrect order with used ascending / descending.
+    */
+    const sortAscending = false;
+    const sortDescending = true;
+
+    test('returns sort number 0 when parameters are not provided', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.naturalSort()).toEqual(0);
+    });
+
+    test('returns sort number -1 when only first paremeter is provided', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.naturalSort('a')).toEqual(-1);
+    });
+
+    test('returns sort number 1 when only second paremeter is provided', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.naturalSort(null, 'b')).toEqual(1);
+    });
+
+    test('returns sort number correctly with string parameters ascending', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.naturalSort('a', 'b', sortAscending)).toEqual(-1);
+    });
+
+    test('returns sort number correctly with string parameters descending', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.naturalSort('a', 'b', sortDescending)).toEqual(1);
+    });
+
+    test('returns sort number correctly with integer parameters ascending', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.naturalSort(1, 2, sortAscending)).toEqual(-1);
+    });
+
+    test('returns sort number correctly with integer parameters descending', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.naturalSort(1, 2, sortDescending)).toEqual(1);
+    });
+
+    test('returns sort number correctly with double parameters ascending', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.naturalSort(10.1, 22.1, sortDescending)).toEqual(1);
+    });
+
+    test('returns sort number correctly with double parameters descending', () => {
+        expect.assertions(1);
+        expect(OskariMock.util.naturalSort(10.1, 22.1, sortAscending)).toEqual(-1);
+    });
```

Comment:
We could add something to test the natural part of the sorting as well like the example from wikipedia:

Alphabetical sorting:

1) z11
2) z2

Natural sorting:

1) z2
2) z11

Reply:
Proposed tests added to test natural part of sorting.